server/routes/prisoner.ts

```diff
@@ -48,7 +49,8 @@ export default function routes(
       }
     }
 
-    const visitSessionData: VisitSessionData = req.session.visitSessionData || { prisoner: undefined }
+    clearSession(req)
+    const visitSessionData: VisitSessionData = req.session.visitSessionData ?? { prisoner: undefined }
```

Comment:
`req.session.visitData` shouldn't actually be present here after the session is cleared; this check is only there to enable `visitSessionData` to be passed through in testing when `clearSession` is mocked out. Couldn't think of a way around this....